src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,102 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    static final int DEFAULT_CAPACITY = 16;
+    static final double LOAD_FACTOR = 0.75;
+
+    static class Node<K, V> {
+        private int hash;
+        private K key;
+        private V value;
+        private Node<K, V> next;
+
+        Node(int hash, K key, V value, Node<K,V> next) {
+            this.hash = hash;
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
+    private Node<K, V>[] buckets;
+    private int hashSize;
+    private int maxSize;
+    private int threshold;
+
+    MyHashMap() {
+        maxSize = DEFAULT_CAPACITY;
+        buckets = (Node<K, V>[]) new Node[maxSize];
+        threshold = (int) (maxSize * LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        checkSize();
+        int keyHash = key == null ? 0 : key.hashCode();
+        int position = Math.abs(keyHash) % maxSize;
+        if (buckets[position] == null) {
+            buckets[position] = new Node<>(keyHash, key, value, null);
+        } else {
+            Node<K,V> pointer = buckets[position];
+            while (pointer.next != null) {
+                if (key == pointer.key || key != null && key.equals(pointer.key)) {
+                    pointer.value = value;
```

Comment:
if i undestand your code correctly , even if hash code not equals u rewrite value ?

Reply:
No, its comparing if objects equals. Its how Objects.equals() actually  comparing two objects.